ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/parallelization/MainnetParallelBlockProcessor.java

```diff
@@ -154,21 +135,48 @@ protected TransactionProcessingResult getTransactionProcessingResult(
     }
   }
 
-  static class ParallelizedPreProcessingContext implements PreprocessingContext {
-    final ParallelizedConcurrentTransactionProcessor parallelizedConcurrentTransactionProcessor;
-
-    public ParallelizedPreProcessingContext(
-        final ParallelizedConcurrentTransactionProcessor
-            parallelizedConcurrentTransactionProcessor) {
-      this.parallelizedConcurrentTransactionProcessor = parallelizedConcurrentTransactionProcessor;
-    }
-
-    public ParallelizedConcurrentTransactionProcessor
-        getParallelizedConcurrentTransactionProcessor() {
-      return parallelizedConcurrentTransactionProcessor;
+  @Override
+  public BlockProcessingResult processBlock(
+      final Blockchain blockchain,
+      final MutableWorldState worldState,
+      final BlockHeader blockHeader,
+      final List<Transaction> transactions,
+      final List<BlockHeader> ommers,
+      final Optional<List<Withdrawal>> maybeWithdrawals,
+      final PrivateMetadataUpdater privateMetadataUpdater) {
+    final BlockProcessingResult blockProcessingResult =
+        super.processBlock(
+            blockchain,
+            worldState,
+            blockHeader,
+            transactions,
+            ommers,
+            maybeWithdrawals,
+            privateMetadataUpdater,
+            new ParallelTransactionPreprocessing());
+    if (blockProcessingResult.isFailed()) {
+      // Fallback to non-parallel processing if there is a block processing exception .
+      LOG.info(
+          "Block processing failed. Falling back to non-parallel processing for block #{} ({})",
```

Comment:
Perhaps change this log to "Parallel transaction processing failure. Falling... ".  Just to be more clear in the log output.

Reply:
modified